### PR TITLE
LogSource: reorder some operations on LogMessage

### PR DESCRIPTION
**Pull request by gaborznagy**
Some mangle callbacks depends on the original LogMessage that is received, therefore their invocation has to be made sooner.

As some processing is elemantary (e.g. hostname resolving), I hesitatedto call it first before anything else to avoid breaking things which assumes differently.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/logsource.c [hunk 1]
```diff
@@ -395,14 +401,18 @@ log_source_queue(LogPipe *s, LogMessage *msg, const LogPathOptions *path_options
 
   log_msg_set_tag_by_id(msg, self->options->source_group_tag);
 
+
+  if (!_invoke_mangle_callbacks(s, msg, path_options))
+    return;
+
+  log_source_override_host(self, msg);
+  log_source_override_program(self, msg);
```
Comment (MrAnno) [inline lib/logsource.c]: I would not extract the `self->options->*_override` conditions, if the names of the extracted functions do not indicate conditionality.

For example,
```C
if (self->options->program_override)
  log_source_override_program(self, msg);
```
is less misleading in my opinion.

**Comment by MrAnno** [comment]
The functionality of `host_override` conflicts with one of our internal mangle callback implementations that wants to use the _original_ hostname.

Moving `log_source_override_host()` after the mangle callback invocation fixes the issue.

**Comment by bazsi** [comment]
I like the extraction into functions, agreeing with the caveat of @MrAnno about naming.

Can't comment on the ordering but it's ok for me either way.

File: lib/logsource.c (hunk 1 above)
Comment (MrAnno) [reply lib/logsource.c]: Thanks.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**